pages/_app.js

```diff
@@ -1,35 +1,55 @@
 import { Footer, Header } from '@scientist-softserv/webstore-component-library'
-import { SessionProvider, signIn, signOut } from 'next-auth/react'
+import { SWRConfig } from 'swr'
+import {
+  SessionProvider, signIn, signOut, useSession
+} from 'next-auth/react'
+
 import {
   FOOTER_NAME,
   FOOTER_SECTIONS,
   FOOTER_SOCIALS,
   LOGO,
   NAVIGATION_LINKS,
-  useCurrentUser,
+  fetcher,
 } from '../utils'
 import '../utils/theme/globals.scss'
 
-// putting the header and footer here mean that they automatically surround every page
-const Webstore = ({ Component, pageProps: { session } }) => {
+const WebStore = ({ Component }) => {
```

Comment:
we can't use `useSession` outside of components that are wrapped by `SessionProvider`. so I moved the actual body of the app into it's own component.

we can now pass the session to the header so we render the "sign in" and "sign out" links appropriately.